src/gui/fileactivitylistmodel.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) by Felix Weilbach <felix.weilbach@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+
+#include "accountstate.h"
+#include "tray/ActivityListModel.h"
+
+namespace OCC {
+
+class FileActivityListModel : public ActivityListModel
+{
+    Q_OBJECT
+
+public:
+    explicit FileActivityListModel(QObject *parent = nullptr);
+
+    Q_INVOKABLE void load(AccountState *accountState, const QString &fileId);
```

Comment:
why not a slot ?
having it return `void` would match the semantic of most slots

Reply:
What is the benefit of using slot instead of Q_INVOKABLE?

Reply:
my rule of thumb is that if something can be a slot it should be one
if it returns a value instead of `void` then I use a normal method added with the `Q_INVOKABLE` method
I find it easier when everything you can call from `qml` is a `slot` because you only need to look at one place and would always keep `Q_INVOKABLE` methods as an exception rather than a rule for easier readability